src/Symfony/Bundle/WebProfilerBundle/Csp/ContentSecurityPolicyHandler.php

```diff
@@ -0,0 +1,253 @@
+            return array(
+                'csp_script_nonce' => $request->headers->get('X-SymfonyProfiler-Script-Nonce'),
+                'csp_style_nonce' => $request->headers->get('X-SymfonyProfiler-Style-Nonce'),
+            );
+        }
+
+        if ($response->headers->has('X-SymfonyProfiler-Script-Nonce') && $response->headers->has('X-SymfonyProfiler-Style-Nonce')) {
+            return array(
+                'csp_script_nonce' => $response->headers->get('X-SymfonyProfiler-Script-Nonce'),
+                'csp_style_nonce' => $response->headers->get('X-SymfonyProfiler-Style-Nonce'),
+            );
+        }
+
+        $nonces = array(
+            'csp_script_nonce' => $this->generateNonce(),
+            'csp_style_nonce' => $this->generateNonce(),
+        );
+
+        $response->headers->set('X-SymfonyProfiler-Script-Nonce', $nonces['csp_script_nonce']);
+        $response->headers->set('X-SymfonyProfiler-Style-Nonce', $nonces['csp_style_nonce']);
+
+        return $nonces;
+    }
+
+    /**
+     * Cleanup temporary headers and updates Content-Security-Policy headers.
+     *
+     * This method should be called on KernelEvents::RESPONSE
+     *
+     * @param Request  $request
+     * @param Response $response
+     *
+     * @return array Nonce used by the bundle in Content-Security-Policy header
+     */
+    public function onKernelResponse(Request $request, Response $response)
+    {
+        $nonces = $this->getNonces($request, $response);
+        $this->cleanHeaders($response);
+        $this->updateCspHeaders($response, $nonces);
+
+        return $nonces;
+    }
+
+    /**
+     * @param Response $response
+     */
+    private function cleanHeaders(Response $response)
+    {
+        $response->headers->remove('X-SymfonyProfiler-Script-Nonce');
+        $response->headers->remove('X-SymfonyProfiler-Style-Nonce');
+    }
+
+    /**
+     * Updates Content-Security-Policy headers in a response.
+     *
+     * @param Response $response
+     * @param array    $nonces
+     *
+     * @return array
+     */
+    private function updateCspHeaders(Response $response, array $nonces = array()) {
+        $nonces = array_replace(array(
+            'csp_script_nonce' => $this->generateNonce(),
+            'csp_style_nonce' => $this->generateNonce(),
+        ), $nonces);
+
+        $ruleIsSet = false;
+
+        $headers = $this->getCspHeaders($response);
+
+        foreach ($headers as $header => $directives) {
+            foreach (array('script-src' => 'csp_script_nonce', 'style-src' => 'csp_style_nonce') as $type => $tokenName) {
+                if (!$this->authorizesInline($directives, $type)) {
+                    if (!isset($headers[$header][$type])) {
+                        if (isset($headers[$header]['default-src'])) {
+                            $headers[$header][$type] = $headers[$header]['default-src'];
```

Comment:
As this is nested pretty deeply, it might be more readable if the first if becomes a guard clause with a continue